### PR TITLE
[enhancement](Nereids) remove deriveStats jobs for some groupExpression 

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes
Don't call DeriveStatJob for follow group expressions:
1. the group expression that is generated by the joinCommute rule
2. the group expression that is generated by the implementation rule without creating a new group

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/ApplyRuleJob.java [hunk 1]
```diff
@@ -82,11 +83,24 @@ public void execute() throws AnalysisException {
...
+                        pushJob(new DeriveStatsJob(newGroupExpression, context));
+                    } else {
+                        // Join Commute rule doesn't change operator's expression and children
+                        // Therefore, the stats can't be changed. Don't derive stats for it
+                        groupExpression.setStatDerived(true);
+                    }
                 } else {
                     pushJob(new CostAndEnforcerJob(newGroupExpression, context));
+                    if (newGroupExpression.children().stream().anyMatch(g -> g.getLogicalExpressions().isEmpty())) {
+                        // Only the implementation rule creates a new group. The groupExpression should be derived
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/ApplyRuleJob.java:96]: “Only the implementation rule creates a new group”  this comments is not always right, please update it

**Review by englefly — DISMISSED** [review]

**Comment by XieJiann** [comment]
run buildall

**Review by englefly — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Pull request merged**